workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -149,33 +136,63 @@ private static void Generate()
                 }
 
                 var workerJsonPath = Path.GetFullPath(Path.Combine(Application.dataPath, ".."));
+                var loggerOutputPath = Path.GetFullPath(Path.Combine(CodegenExeDirectory, "codegen-output.log"));
 
                 using (new ShowProgressBarScope("Generating code..."))
                 {
-                    var errorMessage = new StringBuilder();
                     var exitCode = RedirectedProcess.Command(Common.DotNetBinary)
-                        .WithArgs(ConstructArgs(CodegenExe, schemaCompilerPath, workerJsonPath))
+                        .WithArgs(ConstructArgs(CodegenExe, schemaCompilerPath, workerJsonPath, loggerOutputPath))
                         .RedirectOutputOptions(OutputRedirectBehaviour.None)
-                        .AddErrorProcessing((line) => errorMessage.Append($"\n{line}"))
-                        .AddErrorProcessing(Debug.LogError)
-                        .AddOutputProcessing(ProcessStdOut)
                         .Run();
 
+                    latestCodegenLogs = CodegenLogUtils.ProcessCodegenLogs(loggerOutputPath);
+
+                    var numWarnings = latestCodegenLogs.Count(line => line.level == CodegenLogLevel.WARN);
+                    var numErrors = latestCodegenLogs.Count(line => line.level == CodegenLogLevel.ERROR);
+
                     if (exitCode.ExitCode != 0)
                     {
                         if (!Application.isBatchMode)
                         {
-                            Debug.LogError($"Error(s) compiling schema files!{errorMessage}");
+                            Debug.LogError($"Code generation failed! Please check the code generation logs for more information: {loggerOutputPath}");
                             EditorApplication.delayCall += () =>
                             {
-                                EditorUtility.DisplayDialog("Generate Code",
-                                    "Failed to generate code from schema.\nPlease view the console for errors.",
-                                    "Close");
+                                var option = EditorUtility.DisplayDialogComplex("Generate Code",
+                                    $"Code generation failed with {numWarnings} warnings and {numErrors} errors!\n\nPlease check the code generation logs for more information: {loggerOutputPath}",
+                                    "Open logfile",
+                                    "Close",
+                                    "");
+
+                                switch (option)
+                                {
+                                    // Open logfile
+                                    case 0:
+                                        Application.OpenURL(loggerOutputPath);
```

Comment:
Does this work..? I would expect it to need to be a `file://` URI

Reply:
surprisingly works fine lol

Reply:
Weird. Unity must do some pattern matching on this